Framework/API/inc/MantidAPI/WorkspaceOpOverloads.h

```diff
@@ -54,9 +54,10 @@ MatrixWorkspace_sptr MANTID_API_DLL operator/=(const MatrixWorkspace_sptr &lhs,
 */
 struct MANTID_API_DLL WorkspaceHelpers {
   // Checks whether the binning is the same in two workspaces
-  static bool matchingBins(const MatrixWorkspace &ws1, const MatrixWorkspace &ws2, const bool firstOnly = false);
+  static bool matchingBins(const std::shared_ptr<const MatrixWorkspace> &ws1,
+                           const std::shared_ptr<const MatrixWorkspace> &ws2, const bool firstOnly = false);
```

Comment:
Is there a reason that this file (and the associated `.cpp`) was changed (and resulting in having to change a large number of other files) rather than just dereferencing the pointer in `ConjoinWorkspaces.cpp` [here](https://github.com/mantidproject/mantid/pull/38478/files#diff-a440e18f847529d5cf5108809ba3dfc7fe515a381837f0d1a96e8d982da0faf8R66)?